libraries/stdlib/samples/test/samples/collections/maps.kt

```diff
@@ -318,6 +318,20 @@ class Maps {
             assertPrints(map2, "{beverage=2.7$, meal=12.4$}")
         }
 
+        @Sample
+        fun mapNotNull() {
+            val map1 = mapOf("foo" to 1, "bar" to 2, "baz" to 3)
+            val map2 = map1.mapNotNull {
```

Comment:
There are several problems with this example:
- it doesn't seem to be referenced in code
- `component1` and `equals` are functions to satisfy operator conventions (destructuring and ==), they are rarely called directly in real code. So if you want to compare map entry key with "bar", it would be more idiomatic to write `it.key == "bar"`
- for `if (predicate(it)) it else null` we have `takeIf` and `takeUnless` functions: 
`it.takeUnless { it.key == "bar" }`
- the example overall looks like a contrived way to filter map based on keys, so `Map.filterKeys` would probably do fine in this case. On the other hand, the advantage of `mapNotNull` is that is can additionally transform an element, for example, with it we can filter by keys, but return the corresponding values:
`map1.mapNotNull { entry -> entry.takeUnless { it.key == "bar" }?.value }` 
or using the entry destructuring 
`map1.mapNotNull { (key, value) -> value.takeUnless { key == "bar" } }`